### PR TITLE
Merging MIGraphX ops dialect

### PR DESCRIPTION
**Pull request by jungpark-mlir**

### CODE REVIEW

File: mlir/test/CAPI/ir.c [hunk 1]
```diff
@@ -17,6 +17,7 @@
 #include "mlir-c/BuiltinTypes.h"
 #include "mlir-c/Diagnostics.h"
 #include "mlir-c/Dialect/Standard.h"
+#include "mlir-c/Dialect/MIGraphX.h"
```
Comment (sjw36) [inline mlir/test/CAPI/ir.c]: Is there a reason this was moved and not just copied, adding your code and removing other capi's?

File: mlir/lib/Dialect/MIGraphX/CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,28 @@
+# Configure ROCm support.
```
Comment (sjw36) [inline mlir/lib/Dialect/MIGraphX/CMakeLists.txt]: remove comments

File: mlir/lib/Dialect/MIGraphX/MIGraphXOps.cpp [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+#include "mlir/IR/Value.h"
+#include "mlir/Support/MathExtras.h"
+#include "mlir/Dialect/MIGraphX/MIGraphXOps.h"
+
+#include "mlir/Dialect/MIGraphX/MIGraphXTypes.cpp.inc"
+using namespace mlir;
+//using namespace mlir::migraphx;
+using namespace migraphx;
+
+//===----------------------------------------------------------------------===//
```
Comment (sjw36) [inline mlir/lib/Dialect/MIGraphX/MIGraphXOps.cpp:25]: remove unused unnamed namespace

**Review by sjw36 — CHANGES_REQUESTED** [review]
Overall looks good. Just minor cleanup.
But it looks like some test files were moved out of external/llvm-project. We should keep the llvm code-base as clean and intact as possible. Please keep our capi tests separate and distinct.

File: mlir/test/CAPI/ir.c (hunk 1 above)
Comment (jungpark-mlir) [reply mlir/test/CAPI/ir.c]: Right, I was bit confused. I'll make a separate test

File: mlir/include/mlir/Dialect/MIGraphX/MIGraphXOps.td [hunk 4]
```diff
@@ -0,0 +1,324 @@
...
+// Elementwise binary operations
+
+def MIGraphX_AddOp :
+    MIGraphX_Op<"add">,
+    Arguments<(ins AnyRankedTensor:$inA,
+                   AnyRankedTensor:$inB)>,
+	  Results<(outs AnyRankedTensor:$output)> {	  
+  let summary = "Elementwise binary add";
+  let description = [{
+    Add two tensors elementwise
```
Comment (sjw36) [inline mlir/include/mlir/Dialect/MIGraphX/MIGraphXOps.td:52]: maybe reference the file in migraphx that corresponds to this operation type.

File: mlir/include/mlir/Dialect/MIGraphX/LowerMIGraphXOps.h [hunk 5]
```diff
@@ -0,0 +1,37 @@
+//===- LowerMIGraphXOps.h - MIGraphX conversion ---------------===//
+//
+// Part of the MLIR Project, under the Apache License v2.0 with LLVM Exceptions.
```
Comment (sjw36) [inline mlir/include/mlir/Dialect/MIGraphX/LowerMIGraphXOps.h]: I would like to get away from the Lower*Ops.h that has code in it. For this commit we don't have any lowering steps so let's just leave this out.

**Review by sjw36 — CHANGES_REQUESTED** [review]

File: mlir/include/mlir/Dialect/MIGraphX/LowerMIGraphXOps.h (hunk 5 above)
Comment (jungpark-mlir) [reply mlir/include/mlir/Dialect/MIGraphX/LowerMIGraphXOps.h]: True, got removed.

**Review by sjw36 — APPROVED** [review]

**Pull request merged**